crates/client/rpc-core/src/lib.rs

```diff
@@ -189,3 +208,16 @@ pub trait StarknetTraceRpcApi {
     /// Returns the execution trace of a transaction
     async fn trace_transaction(&self, transaction_hash: FieldElement) -> RpcResult<TransactionTrace>;
 }
+
+pub enum DeclareTransactionCommonInput {
+    V0(DeclareTransactionV0V1, Vec<u8>, IndexMap<EntryPointType, Vec<EntryPoint>>, usize),
+    V1(DeclareTransaction),
+}
+
+#[async_trait]
+pub trait StarknetRpcApiCommonFuncs {
```

Comment:
I was thinking we can skip the trait for this altogether and just do `impl Starknet`. The reason was that we already have other implementations which are commonly accessed and just want this to move to a new impl block because it has different constraints on the generics